Add Response::redirect

jbr opened the pull request:
The Redirect endpoint is a special case ("always redirect") of the broader need to conditionally redirect based on some logic within a route handler. This PR moves the creation of a temporary redirect into Response and uses that in the Redirect endpoint. Accidentally, this also eliminates a string clone, since set_header only needs a &str.

prabirshrestha commented:
Might be better to have redirect_permanent and redirect_temporary to be explicit?

jbr commented:
Agreed, and that also seems like a distinction the `Redirect` endpoint will want to make eventually. I tried to keep the changes small and focused in this PR, but will happily add a second fn. Also considered a second argument with an enum or boolean, but that seems ungainly

yoshuawuyts commented on src/response/mod.rs:
```diff
@@ -53,6 +53,12 @@ impl Response {
         }
     }
 
+    /// Creates a response that represents a temporary redirect to `location`.
```
Could we add an example here?

yoshuawuyts commented:
@jbr I like what you're going for with this PR. I think following @prabirshrestha's suggestion to implement both versions might just be easiest; Tide is still in a place where we can issue breaking changes frequently so if we know which API we want, we can implement it.

So my understanding is this would be:
```rust
fn redirect_permanent;
fn redirect_temporary;

impl Response {
    fn redirect_permanent;
    fn redirect_temporary;
}
```

I wonder if it would make sense to shorten the suffix here:
```rust
fn redirect_perm;
fn redirect_temp;

impl Response {
    fn redirect_perm;
    fn redirect_temp;
}
```

jbr commented:
Addressed:
+ Added redirect_perm
+ Renamed redirect to redirect_temp
+ Added [hopefully] relatively realistic examples for each

Not [yet] addressed in this PR: Making the same change to the Redirect endpoint / adding a permanent redirect endpoint
